src/main/java/com/snowflake/kafka/connector/internal/streaming/SnowflakeSinkServiceV2.java

```diff
@@ -537,8 +549,7 @@ public static String partitionChannelKey(String topic, int partition) {
   /* Used for testing */
   @VisibleForTesting
   public SnowflakeStreamingIngestClient getStreamingIngestClient() {
-    return StreamingClientProvider.getStreamingClientProviderInstance()
-        .getClient(this.connectorConfig);
+    return this.streamingIngestClient;
```

Comment:
Is this correct? I assume we want to get it from the client provider?

Reply:
When client optimization is off calling `.getClient()` implicitly creates a new client instance. This is very misleading in tests, because I cannot verify that the client was actually closed.

In general provider abstraction is tricky when optimization is off.